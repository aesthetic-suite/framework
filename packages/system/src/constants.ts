import { ColorShade, ElevationType, PaletteType } from './types';

export const DEPTHS: Record<ElevationType, number> = {
  content: 100, // xs
  navigation: 1000, // sm
  sheet: 1100, // lg
  overlay: 1200, // lg
  modal: 1300, // xl
  toast: 1400, // md
  dialog: 1500, // md
  menu: 1600, // md
  tooltip: 1700, // sm
};

export const SHADE_RANGES: ColorShade[] = [
  '00',
  '10',
  '20',
  '30',
  '40',
  '50',
  '60',
  '70',
  '80',
  '90',
];

export const PALETTE_TYPES: PaletteType[] = [
  'brand',
  'primary',
  'secondary',
  'tertiary',
  'neutral',
  'muted',
  'danger',
  'warning',
  'negative',
  'positive',
];
