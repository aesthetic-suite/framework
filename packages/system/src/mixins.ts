import { objectLoop } from '@aesthetic/utils';
import { MixinTemplate, MixinType, Utilities } from './types';

export function hideOffscreen() {
	return {
		clipPath: 'rect(1px, 1px, 1px, 1px)',
		height: 1,
		left: '-5vw',
		overflow: 'hidden',
		position: 'fixed',
		width: 1,
	};
}

export function hideVisually() {
	return {
		'@selectors': {
			':not(:focus):not(:active)': {
				border: 0,
				clip: 'rect(0, 0, 0, 0)',
				clipPath: 'inset(50%)',
				height: 1,
				margin: 0,
				overflow: 'hidden',
				padding: 0,
				position: 'absolute',
				whiteSpace: 'nowrap',
				width: 1,
			},
		},
	};
}

export function resetButton() {
	return {
		appearance: 'none',
		backgroundColor: 'transparent',
		border: 0,
		cursor: 'pointer',
		display: 'inline-flex',
		fontSize: 'inherit',
		margin: 0,
		padding: 0,
		textDecoration: 'none',
		userSelect: 'auto',
		verticalAlign: 'middle',
	};
}

export function resetInput() {
	return {
		appearance: 'none',
		backgroundColor: 'transparent',
		margin: 0,
		padding: 0,
		width: '100%',
		'@selectors': {
			'::-moz-focus-outer': {
				border: 0,
			},
		},
	};
}

export function resetList() {
	return {
		listStyle: 'none',
		margin: 0,
		padding: 0,
	};
}

export function resetMedia() {
	return {
		display: 'block',
		verticalAlign: 'middle',
	};
}

export function resetTypography() {
	return {
		fontFamily: 'inherit',
		fontSize: 'inherit',
		fontWeight: 'normal',
		wordWrap: 'break-word',
	};
}

export function root(css: Utilities<object>) {
	const declaration = {
		backgroundColor: css.var('palette-neutral-color-00'),
		color: css.var('palette-neutral-text'),
		fontFamily: css.var('typography-font-text'),
		fontSize: css.var('typography-root-text-size'),
		lineHeight: css.var('typography-root-line-height'),
		textRendering: 'optimizeLegibility',
		textSizeAdjust: '100%',
		margin: 0,
		padding: 0,
		'-webkit-font-smoothing': 'antialiased',
		'-moz-osx-font-smoothing': 'grayscale',
		'@media': {},
	};

	// Fluid typography!
	objectLoop(css.tokens.breakpoint, (bp, size) => {
		(declaration['@media'] as Record<string, object>)[bp.query] = {
			fontSize: css.var(`breakpoint-${size}-root-text-size`),
			lineHeight: css.var(`breakpoint-${size}-root-line-height`),
		};
	});

	return declaration;
}

export function textBreak() {
	return {
		overflowWrap: 'break-word',
		wordWrap: 'break-word',
		wordBreak: 'break-word',
	};
}

export function textTruncate() {
	return {
		overflow: 'hidden',
		textOverflow: 'ellipsis',
		whiteSpace: 'nowrap',
	};
}

export function textWrap() {
	return {
		overflowWrap: 'normal',
		wordWrap: 'normal',
		wordBreak: 'normal',
	};
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const MIXIN_MAP: Record<MixinType, MixinTemplate<any>> = {
	'hide-offscreen': hideOffscreen,
	'hide-visually': hideVisually,
	'reset-button': resetButton,
	'reset-input': resetInput,
	'reset-list': resetList,
	'reset-media': resetMedia,
	'reset-typography': resetTypography,
	root,
	'text-break': textBreak,
	'text-truncate': textTruncate,
	'text-wrap': textWrap,
};
