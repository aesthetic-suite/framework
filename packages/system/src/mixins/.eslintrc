{
  "rules": {
    "babel/no-invalid-this": "off"
  }
}
