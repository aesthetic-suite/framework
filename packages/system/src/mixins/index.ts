import { border } from './border';
import { box } from './box';
import { heading } from './heading';
import {
  hideCompletely,
  hideOffscreen,
  hideVisually,
  resetButton,
  resetInput,
  resetList,
  resetMedia,
  resetTypography,
  root,
  textBreak,
  textTruncate,
  textWrap,
} from './pattern';
import { shadow } from './shadow';
import { text } from './text';

export default {
  border,
  box,
  heading,
  hideCompletely,
  hideOffscreen,
  hideVisually,
  resetButton,
  resetInput,
  resetList,
  resetMedia,
  resetTypography,
  root,
  shadow,
  text,
  textBreak,
  textTruncate,
  textWrap,
};
