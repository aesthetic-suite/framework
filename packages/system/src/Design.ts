import deepMerge from 'extend';
import Theme from './Theme';
import { DEPTHS } from './constants';
import { DesignTokens, DeepPartial, ThemeOptions, ThemeTokens } from './types';

export default class Design {
  readonly rootLineHeight: number;

  readonly rootTextSize: number;

  readonly spacingUnit: number;

  readonly tokens: DesignTokens;

  constructor(tokens: Omit<DesignTokens, 'depth' | 'unit'>) {
    this.tokens = {
      ...tokens,
      depth: DEPTHS,
    };

    this.rootLineHeight = tokens.typography.rootLineHeight;
    this.rootTextSize = parseFloat(tokens.typography.rootTextSize);
    this.spacingUnit =
      tokens.spacing.type === 'vertical-rhythm'
        ? this.rootTextSize * this.rootLineHeight
        : tokens.spacing.unit;
  }

  /**
   * Create a new theme with the defined theme tokens, while inheriting the shared design tokens.
   */
  createTheme(options: ThemeOptions, tokens: ThemeTokens): Theme {
    return new Theme(options, tokens, this);
  }

  /**
   * Extend and instantiate a new design instance with customized design tokens.
   */
  extend(tokens: DeepPartial<DesignTokens>): Design {
    return new Design(deepMerge(true, {}, this.tokens, tokens));
  }
}
