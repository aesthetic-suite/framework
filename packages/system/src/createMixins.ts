import { objectCreate } from '@aesthetic/utils';
import mixins from './mixins';
import { Mixins, VarUtil, Tokens } from './types';
import { BORDER_SIZES, HEADING_SIZES, SHADOW_SIZES, TEXT_SIZES, PALETTE_TYPES } from './constants';

export default function createMixins(
  vars: VarUtil,
  // Only ask for what we need
  tokens: {
    breakpoint: Tokens['breakpoint'];
  },
): Mixins {
  return {
    background: objectCreate(PALETTE_TYPES, (palette) => mixins.background(vars, palette)),
    border: objectCreate(BORDER_SIZES, (size) => mixins.border(vars, size)),
    box: objectCreate(TEXT_SIZES, (size) => mixins.box(vars, size)),
    foreground: objectCreate(PALETTE_TYPES, (palette) => mixins.foreground(vars, palette)),
    heading: objectCreate(HEADING_SIZES, (size) => mixins.heading(vars, size)),
    pattern: {
      hide: {
        completely: mixins.hideCompletely(),
        offscreen: mixins.hideOffscreen(),
        visually: mixins.hideVisually(),
      },
      reset: {
        button: mixins.resetButton(),
        input: mixins.resetInput(),
        list: mixins.resetList(),
        media: mixins.resetMedia(),
        typography: mixins.resetTypography(),
      },
      root: mixins.root(vars, tokens.breakpoint),
      text: {
        break: mixins.textBreak(),
        truncate: mixins.textTruncate(),
        wrap: mixins.textWrap(),
      },
    },
    shadow: objectCreate(SHADOW_SIZES, (size) => mixins.shadow(vars, size)),
    text: objectCreate(TEXT_SIZES, (size) => mixins.text(vars, size)),
    ui: {
      button: objectCreate(PALETTE_TYPES, (palette) => mixins.button(vars, palette)),
    },
  };
}
