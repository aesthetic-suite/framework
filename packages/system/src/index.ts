/**
 * @copyright   2017-2019, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 */

import Design from './Design';
import Theme from './Theme';

export * from './types';

export { Design, Theme };
