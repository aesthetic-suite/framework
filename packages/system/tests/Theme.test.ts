/* eslint-disable babel/no-invalid-this */

import { Rule } from '@aesthetic/types';
import { Theme, Utilities } from '../src';
import { design, lightTheme } from '../src/test';

interface TextOptions {
  size?: string;
}

function text(this: Utilities<Rule>, { size = 'df' }: TextOptions) {
  return {
    color: this.var('palette-neutral-fg-base'),
    fontFamily: this.var('typography-font-text'),
    fontSize: this.var(`text-${size}-size` as 'text-df-size'),
    lineHeight: this.var(`text-${size}-line-height` as 'text-df-line-height'),
  };
}

describe('Theme', () => {
  let testTheme: Theme<Rule>;

  beforeEach(() => {
    // Copy so we dont mutate the testing mock
    testTheme = lightTheme.extend({});
    testTheme.registerMixin('text', text);
  });

  it('sets class properties', () => {
    expect(testTheme.contrast).toBe('normal');
    expect(testTheme.scheme).toBe('light');
  });

  it('can extend and create a new theme', () => {
    const newTheme = testTheme.extend(
      {
        palette: {
          brand: {
            color: {
              '00': 'red',
            },
          },
        },
      },
      { contrast: 'high' },
    );

    expect(newTheme).toBeInstanceOf(Theme);
    expect(newTheme.contrast).toBe('high');
    expect(newTheme.scheme).toBe('light');
    expect(newTheme.tokens.palette.brand.color['00']).toBe('red');
  });

  it('registers built-ins from parent design on instantiation', () => {
    const theme = new Theme(
      { contrast: 'normal', scheme: 'light' },
      lightTheme.tokens,
      design.extend('test', {}, { text }),
    );

    expect(theme.mixin).toHaveProperty('text');
    expect((theme.mixin as any).text()).toMatchSnapshot();
  });

  describe('extendMixin()', () => {
    it('creates a template set for the defined name', () => {
      // @ts-expect-error Allow access
      expect(testTheme.templates.test).toBeUndefined();

      testTheme.extendMixin('test', () => ({}));

      // @ts-expect-error Allow access
      expect(testTheme.templates.test).toBeInstanceOf(Set);
    });

    it('doesnt set the same template twice', () => {
      const template = () => ({});

      testTheme.extendMixin('test', template);
      testTheme.extendMixin('test', template);
      testTheme.extendMixin('test', template);

      // @ts-expect-error Allow access
      expect(testTheme.templates.test.size).toBe(1);
    });

    it('can extend a mixin with additional properties', () => {
      expect(testTheme.mixin('text')).toMatchSnapshot();

      testTheme.extendMixin('text', () => ({
        fontWeight: 'bold',
        color: 'red',
        display: 'inline-block',
        opacity: 1,
      }));

      expect(testTheme.mixin('text')).toMatchSnapshot();
    });

    it('can utilize the same options as the parent mixin', () => {
      testTheme.extendMixin('text', ({ size }: TextOptions) => ({
        fontWeight: size === 'lg' ? 'bold' : 'normal',
      }));

      expect(testTheme.mixin('text')).toMatchSnapshot();
      expect(testTheme.mixin('text', { size: 'lg' }, {})).toMatchSnapshot();
    });
  });

  describe('mixin()', () => {
    it('returns an empty object for an unknown mixin', () => {
      const spy = jest.spyOn(console, 'warn').mockImplementation();

      expect(testTheme.mixin('unknown-mixin')).toEqual({});
      expect(spy).toHaveBeenCalledWith('Unknown mixin "unknown-mixin".');

      spy.mockRestore();
    });

    it('returns base properties when no options or extra rule passed', () => {
      expect(testTheme.mixin('text')).toMatchSnapshot();
    });

    it('returns merged properties when no options passed but an extra rule is passed', () => {
      expect(
        testTheme.mixin('text', {
          fontSize: '60px',
        }),
      ).toMatchSnapshot();
    });

    it('returns merged and customized properties when options and extra rule passed', () => {
      expect(
        testTheme.mixin(
          'text',
          { size: 'large' },
          {
            fontSize: '60px',
          },
        ),
      ).toMatchSnapshot();
    });

    it('merges multiple templates with base mixin', () => {
      testTheme.extendMixin('text', () => ({ fontWeight: 'bold' }));
      testTheme.extendMixin('text', () => ({ fontSize: '24px' }));

      expect(testTheme.mixin('text')).toMatchSnapshot();
    });

    it('merges templates and rules with base mixin', () => {
      testTheme.extendMixin('text', () => ({ fontWeight: 'bold' }));

      expect(testTheme.mixin('text', {}, { fontSize: '24px' })).toMatchSnapshot();
    });

    it('allows additional rules to overwrite templates and mixin', () => {
      testTheme.extendMixin('text', () => ({ fontWeight: 'bold' }));

      expect(testTheme.mixin('text', {}, { fontWeight: 800 })).toMatchSnapshot();
    });
  });

  describe('registerMixin()', () => {
    it('errors if trying to overwrite an existing mixin', () => {
      expect(() => {
        testTheme.registerMixin('text', () => ({}));
        testTheme.registerMixin('text', () => ({}));
      }).toThrow('A mixin already exists for "text". Cannot overwrite.');
    });

    it('registers a custom mixin', () => {
      // @ts-expect-error Allow access
      expect(testTheme.mixins.custom).toBeUndefined();

      const template = () => ({});
      testTheme.registerMixin('custom', template);

      // @ts-expect-error Allow access
      expect(testTheme.mixins.custom).toBe(template);
    });
  });

  describe('toVariables()', () => {
    it('returns tokens as a flat object of CSS variables', () => {
      expect(testTheme.toVariables()).toMatchSnapshot();
    });

    it('caches and returns same instance', () => {
      const a = testTheme.toVariables();
      const b = testTheme.toVariables();

      expect(a).toBe(b);
    });
  });

  describe('unit()', () => {
    it('generates a rem unit', () => {
      expect(testTheme.unit(-3.15)).toBe('-3.94rem');
      expect(testTheme.unit(1)).toBe('1.25rem');
      expect(testTheme.unit(6.25)).toBe('7.81rem');
    });

    it('supports multiple values', () => {
      expect(testTheme.unit(1, 2, 3, 4)).toBe('1.25rem 2.50rem 3.75rem 5rem');
    });
  });

  describe('var()', () => {
    it('returns a CSS var', () => {
      expect(testTheme.var('breakpoint-lg-root-line-height')).toBe(
        'var(--breakpoint-lg-root-line-height)',
      );
    });

    it('supports multiple fallbacks', () => {
      expect(testTheme.var('border-lg-radius', testTheme.var('border-df-radius'), '10px')).toBe(
        'var(--border-lg-radius, var(--border-df-radius), 10px)',
      );
    });
  });
});
