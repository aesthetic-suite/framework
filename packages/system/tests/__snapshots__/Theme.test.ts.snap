// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme extendMixin() can extend a mixing with additional properties 1`] = `
Object {
  "color": "var(--palette-neutral-fg-base)",
  "fontFamily": "var(--typography-font-text)",
  "fontSize": "var(--text-df-size)",
  "fontWeight": "normal",
  "lineHeight": "var(--text-df-line-height)",
  "wordWrap": "break-word",
}
`;

exports[`Theme extendMixin() can extend a mixing with additional properties 2`] = `
Object {
  "color": "red",
  "display": "inline-block",
  "fontFamily": "var(--typography-font-text)",
  "fontSize": "var(--text-df-size)",
  "fontWeight": "bold",
  "lineHeight": "var(--text-df-line-height)",
  "opacity": 1,
  "wordWrap": "break-word",
}
`;

exports[`Theme extendMixin() will deep merge properties 1`] = `
Object {
  "borderColor": "var(--palette-neutral-color-40)",
  "borderRadius": "var(--border-df-radius)",
  "borderStyle": "solid",
  "borderWidth": "var(--border-df-width)",
}
`;

exports[`Theme extendMixin() will deep merge properties 2`] = `
Object {
  ":focus": Object {
    "outline": "none",
  },
  "borderColor": "var(--palette-neutral-color-40)",
  "borderRadius": "var(--border-df-radius)",
  "borderStyle": "solid",
  "borderWidth": "var(--border-df-width)",
}
`;

exports[`Theme mixin() can merge from multiple targets 1`] = `
Object {
  "backgroundColor": "var(--palette-neutral-bg-base)",
  "borderColor": "var(--palette-neutral-color-40)",
  "borderRadius": "var(--border-sm-radius)",
  "borderStyle": "solid",
  "borderWidth": "var(--border-sm-width)",
  "color": "red",
  "fontFamily": "var(--typography-font-text)",
  "fontSize": "var(--text-df-size)",
  "fontWeight": "normal",
  "letterSpacing": "var(--heading-l3-letter-spacing)",
  "lineHeight": "var(--text-df-line-height)",
  "wordWrap": "break-word",
}
`;

exports[`Theme mixin() merges objects with mixin 1`] = `
Object {
  "color": "var(--palette-neutral-fg-base)",
  "fontFamily": "var(--typography-font-text)",
  "fontSize": "24px",
  "fontWeight": "bold",
  "lineHeight": "var(--text-df-line-height)",
  "wordWrap": "break-word",
}
`;

exports[`Theme overwriteMixin() can completely replace an existing mixin 1`] = `
Object {
  "color": "var(--palette-neutral-fg-base)",
  "fontFamily": "var(--typography-font-text)",
  "fontSize": "var(--text-df-size)",
  "fontWeight": "normal",
  "lineHeight": "var(--text-df-line-height)",
  "wordWrap": "break-word",
}
`;

exports[`Theme overwriteMixin() can completely replace an existing mixin 2`] = `
Object {
  "fontSize": "20px",
}
`;

exports[`Theme toVariables() returns tokens as a flat object of CSS variables 1`] = `
Object {
  "border-df-radius": "0.21rem",
  "border-df-width": "0.07rem",
  "border-lg-radius": "0.32rem",
  "border-lg-width": "0.14rem",
  "border-sm-radius": "0.14rem",
  "border-sm-width": "0.04rem",
  "breakpoint-lg-query": "(min-width: 114.29em) and (max-width: 137.07em)",
  "breakpoint-lg-query-size": 1600,
  "breakpoint-lg-root-line-height": 1.62,
  "breakpoint-lg-root-text-size": "18.15px",
  "breakpoint-md-query": "(min-width: 91.43em) and (max-width: 114.21em)",
  "breakpoint-md-query-size": 1280,
  "breakpoint-md-root-line-height": 1.52,
  "breakpoint-md-root-text-size": "17.01px",
  "breakpoint-sm-query": "(min-width: 68.57em) and (max-width: 91.36em)",
  "breakpoint-sm-query-size": 960,
  "breakpoint-sm-root-line-height": 1.42,
  "breakpoint-sm-root-text-size": "15.94px",
  "breakpoint-xl-query": "(min-width: 137.14em)",
  "breakpoint-xl-query-size": 1920,
  "breakpoint-xl-root-line-height": 1.73,
  "breakpoint-xl-root-text-size": "19.36px",
  "breakpoint-xs-query": "(min-width: 45.71em) and (max-width: 68.50em)",
  "breakpoint-xs-query-size": 640,
  "breakpoint-xs-root-line-height": 1.33,
  "breakpoint-xs-root-text-size": "14.94px",
  "depth-content": 100,
  "depth-dialog": 1500,
  "depth-menu": 1600,
  "depth-modal": 1300,
  "depth-navigation": 1000,
  "depth-overlay": 1200,
  "depth-sheet": 1100,
  "depth-toast": 1400,
  "depth-tooltip": 1700,
  "heading-l1-letter-spacing": "0.25px",
  "heading-l1-line-height": 1.5,
  "heading-l1-size": "1.14rem",
  "heading-l2-letter-spacing": "0.33px",
  "heading-l2-line-height": 1.69,
  "heading-l2-size": "1.43rem",
  "heading-l3-letter-spacing": "0.44px",
  "heading-l3-line-height": 1.9,
  "heading-l3-size": "1.79rem",
  "heading-l4-letter-spacing": "0.59px",
  "heading-l4-line-height": 2.14,
  "heading-l4-size": "2.23rem",
  "heading-l5-letter-spacing": "0.79px",
  "heading-l5-line-height": 2.4,
  "heading-l5-size": "2.79rem",
  "heading-l6-letter-spacing": "1.05px",
  "heading-l6-line-height": 2.7,
  "heading-l6-size": "3.49rem",
  "palette-brand-bg-base": "#fff",
  "palette-brand-bg-disabled": "#fff",
  "palette-brand-bg-focused": "#fff",
  "palette-brand-bg-hovered": "#fff",
  "palette-brand-bg-selected": "#fff",
  "palette-brand-color-00": "#fff",
  "palette-brand-color-10": "#fff",
  "palette-brand-color-20": "#fff",
  "palette-brand-color-30": "#fff",
  "palette-brand-color-40": "#fff",
  "palette-brand-color-50": "#fff",
  "palette-brand-color-60": "#fff",
  "palette-brand-color-70": "#fff",
  "palette-brand-color-80": "#fff",
  "palette-brand-color-90": "#fff",
  "palette-brand-fg-base": "#fff",
  "palette-brand-fg-disabled": "#fff",
  "palette-brand-fg-focused": "#fff",
  "palette-brand-fg-hovered": "#fff",
  "palette-brand-fg-selected": "#fff",
  "palette-danger-bg-base": "#fff",
  "palette-danger-bg-disabled": "#fff",
  "palette-danger-bg-focused": "#fff",
  "palette-danger-bg-hovered": "#fff",
  "palette-danger-bg-selected": "#fff",
  "palette-danger-color-00": "#fff",
  "palette-danger-color-10": "#fff",
  "palette-danger-color-20": "#fff",
  "palette-danger-color-30": "#fff",
  "palette-danger-color-40": "#fff",
  "palette-danger-color-50": "#fff",
  "palette-danger-color-60": "#fff",
  "palette-danger-color-70": "#fff",
  "palette-danger-color-80": "#fff",
  "palette-danger-color-90": "#fff",
  "palette-danger-fg-base": "#fff",
  "palette-danger-fg-disabled": "#fff",
  "palette-danger-fg-focused": "#fff",
  "palette-danger-fg-hovered": "#fff",
  "palette-danger-fg-selected": "#fff",
  "palette-failure-bg-base": "#fff",
  "palette-failure-bg-disabled": "#fff",
  "palette-failure-bg-focused": "#fff",
  "palette-failure-bg-hovered": "#fff",
  "palette-failure-bg-selected": "#fff",
  "palette-failure-color-00": "#fff",
  "palette-failure-color-10": "#fff",
  "palette-failure-color-20": "#fff",
  "palette-failure-color-30": "#fff",
  "palette-failure-color-40": "#fff",
  "palette-failure-color-50": "#fff",
  "palette-failure-color-60": "#fff",
  "palette-failure-color-70": "#fff",
  "palette-failure-color-80": "#fff",
  "palette-failure-color-90": "#fff",
  "palette-failure-fg-base": "#fff",
  "palette-failure-fg-disabled": "#fff",
  "palette-failure-fg-focused": "#fff",
  "palette-failure-fg-hovered": "#fff",
  "palette-failure-fg-selected": "#fff",
  "palette-info-bg-base": "#fff",
  "palette-info-bg-disabled": "#fff",
  "palette-info-bg-focused": "#fff",
  "palette-info-bg-hovered": "#fff",
  "palette-info-bg-selected": "#fff",
  "palette-info-color-00": "#fff",
  "palette-info-color-10": "#fff",
  "palette-info-color-20": "#fff",
  "palette-info-color-30": "#fff",
  "palette-info-color-40": "#fff",
  "palette-info-color-50": "#fff",
  "palette-info-color-60": "#fff",
  "palette-info-color-70": "#fff",
  "palette-info-color-80": "#fff",
  "palette-info-color-90": "#fff",
  "palette-info-fg-base": "#fff",
  "palette-info-fg-disabled": "#fff",
  "palette-info-fg-focused": "#fff",
  "palette-info-fg-hovered": "#fff",
  "palette-info-fg-selected": "#fff",
  "palette-muted-bg-base": "#fff",
  "palette-muted-bg-disabled": "#fff",
  "palette-muted-bg-focused": "#fff",
  "palette-muted-bg-hovered": "#fff",
  "palette-muted-bg-selected": "#fff",
  "palette-muted-color-00": "#fff",
  "palette-muted-color-10": "#fff",
  "palette-muted-color-20": "#fff",
  "palette-muted-color-30": "#fff",
  "palette-muted-color-40": "#fff",
  "palette-muted-color-50": "#fff",
  "palette-muted-color-60": "#fff",
  "palette-muted-color-70": "#fff",
  "palette-muted-color-80": "#fff",
  "palette-muted-color-90": "#fff",
  "palette-muted-fg-base": "#fff",
  "palette-muted-fg-disabled": "#fff",
  "palette-muted-fg-focused": "#fff",
  "palette-muted-fg-hovered": "#fff",
  "palette-muted-fg-selected": "#fff",
  "palette-neutral-bg-base": "#fff",
  "palette-neutral-bg-disabled": "#fff",
  "palette-neutral-bg-focused": "#fff",
  "palette-neutral-bg-hovered": "#fff",
  "palette-neutral-bg-selected": "#fff",
  "palette-neutral-color-00": "#fff",
  "palette-neutral-color-10": "#fff",
  "palette-neutral-color-20": "#fff",
  "palette-neutral-color-30": "#fff",
  "palette-neutral-color-40": "#fff",
  "palette-neutral-color-50": "#fff",
  "palette-neutral-color-60": "#fff",
  "palette-neutral-color-70": "#fff",
  "palette-neutral-color-80": "#fff",
  "palette-neutral-color-90": "#fff",
  "palette-neutral-fg-base": "#fff",
  "palette-neutral-fg-disabled": "#fff",
  "palette-neutral-fg-focused": "#fff",
  "palette-neutral-fg-hovered": "#fff",
  "palette-neutral-fg-selected": "#fff",
  "palette-primary-bg-base": "#fff",
  "palette-primary-bg-disabled": "#fff",
  "palette-primary-bg-focused": "#fff",
  "palette-primary-bg-hovered": "#fff",
  "palette-primary-bg-selected": "#fff",
  "palette-primary-color-00": "#fff",
  "palette-primary-color-10": "#fff",
  "palette-primary-color-20": "#fff",
  "palette-primary-color-30": "#fff",
  "palette-primary-color-40": "#fff",
  "palette-primary-color-50": "#fff",
  "palette-primary-color-60": "#fff",
  "palette-primary-color-70": "#fff",
  "palette-primary-color-80": "#fff",
  "palette-primary-color-90": "#fff",
  "palette-primary-fg-base": "#fff",
  "palette-primary-fg-disabled": "#fff",
  "palette-primary-fg-focused": "#fff",
  "palette-primary-fg-hovered": "#fff",
  "palette-primary-fg-selected": "#fff",
  "palette-secondary-bg-base": "#fff",
  "palette-secondary-bg-disabled": "#fff",
  "palette-secondary-bg-focused": "#fff",
  "palette-secondary-bg-hovered": "#fff",
  "palette-secondary-bg-selected": "#fff",
  "palette-secondary-color-00": "#fff",
  "palette-secondary-color-10": "#fff",
  "palette-secondary-color-20": "#fff",
  "palette-secondary-color-30": "#fff",
  "palette-secondary-color-40": "#fff",
  "palette-secondary-color-50": "#fff",
  "palette-secondary-color-60": "#fff",
  "palette-secondary-color-70": "#fff",
  "palette-secondary-color-80": "#fff",
  "palette-secondary-color-90": "#fff",
  "palette-secondary-fg-base": "#fff",
  "palette-secondary-fg-disabled": "#fff",
  "palette-secondary-fg-focused": "#fff",
  "palette-secondary-fg-hovered": "#fff",
  "palette-secondary-fg-selected": "#fff",
  "palette-success-bg-base": "#fff",
  "palette-success-bg-disabled": "#fff",
  "palette-success-bg-focused": "#fff",
  "palette-success-bg-hovered": "#fff",
  "palette-success-bg-selected": "#fff",
  "palette-success-color-00": "#fff",
  "palette-success-color-10": "#fff",
  "palette-success-color-20": "#fff",
  "palette-success-color-30": "#fff",
  "palette-success-color-40": "#fff",
  "palette-success-color-50": "#fff",
  "palette-success-color-60": "#fff",
  "palette-success-color-70": "#fff",
  "palette-success-color-80": "#fff",
  "palette-success-color-90": "#fff",
  "palette-success-fg-base": "#fff",
  "palette-success-fg-disabled": "#fff",
  "palette-success-fg-focused": "#fff",
  "palette-success-fg-hovered": "#fff",
  "palette-success-fg-selected": "#fff",
  "palette-tertiary-bg-base": "#fff",
  "palette-tertiary-bg-disabled": "#fff",
  "palette-tertiary-bg-focused": "#fff",
  "palette-tertiary-bg-hovered": "#fff",
  "palette-tertiary-bg-selected": "#fff",
  "palette-tertiary-color-00": "#fff",
  "palette-tertiary-color-10": "#fff",
  "palette-tertiary-color-20": "#fff",
  "palette-tertiary-color-30": "#fff",
  "palette-tertiary-color-40": "#fff",
  "palette-tertiary-color-50": "#fff",
  "palette-tertiary-color-60": "#fff",
  "palette-tertiary-color-70": "#fff",
  "palette-tertiary-color-80": "#fff",
  "palette-tertiary-color-90": "#fff",
  "palette-tertiary-fg-base": "#fff",
  "palette-tertiary-fg-disabled": "#fff",
  "palette-tertiary-fg-focused": "#fff",
  "palette-tertiary-fg-hovered": "#fff",
  "palette-tertiary-fg-selected": "#fff",
  "palette-warning-bg-base": "#fff",
  "palette-warning-bg-disabled": "#fff",
  "palette-warning-bg-focused": "#fff",
  "palette-warning-bg-hovered": "#fff",
  "palette-warning-bg-selected": "#fff",
  "palette-warning-color-00": "#fff",
  "palette-warning-color-10": "#fff",
  "palette-warning-color-20": "#fff",
  "palette-warning-color-30": "#fff",
  "palette-warning-color-40": "#fff",
  "palette-warning-color-50": "#fff",
  "palette-warning-color-60": "#fff",
  "palette-warning-color-70": "#fff",
  "palette-warning-color-80": "#fff",
  "palette-warning-color-90": "#fff",
  "palette-warning-fg-base": "#fff",
  "palette-warning-fg-disabled": "#fff",
  "palette-warning-fg-focused": "#fff",
  "palette-warning-fg-hovered": "#fff",
  "palette-warning-fg-selected": "#fff",
  "shadow-lg-blur": "0.77rem",
  "shadow-lg-spread": "0rem",
  "shadow-lg-x": "0rem",
  "shadow-lg-y": "0.61rem",
  "shadow-md-blur": "0.44rem",
  "shadow-md-spread": "0rem",
  "shadow-md-x": "0rem",
  "shadow-md-y": "0.37rem",
  "shadow-sm-blur": "0.25rem",
  "shadow-sm-spread": "0rem",
  "shadow-sm-x": "0rem",
  "shadow-sm-y": "0.23rem",
  "shadow-xl-blur": "1.34rem",
  "shadow-xl-spread": "0rem",
  "shadow-xl-x": "0rem",
  "shadow-xl-y": "0.98rem",
  "shadow-xs-blur": "0.14rem",
  "shadow-xs-spread": "0rem",
  "shadow-xs-x": "0rem",
  "shadow-xs-y": "0.14rem",
  "spacing-df": "1.25rem",
  "spacing-lg": "3.75rem",
  "spacing-md": "2.50rem",
  "spacing-sm": "0.63rem",
  "spacing-type": "vertical-rhythm",
  "spacing-unit": 17.5,
  "spacing-xl": "5rem",
  "spacing-xs": "0.31rem",
  "text-df-line-height": 1.25,
  "text-df-size": "1rem",
  "text-lg-line-height": 1.25,
  "text-lg-size": "1.13rem",
  "text-sm-line-height": 1.25,
  "text-sm-size": "0.89rem",
  "typography-font-heading": "system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, Helvetica, Arial, sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\"",
  "typography-font-monospace": "\\"Lucida Console\\", Monaco, monospace",
  "typography-font-system": "system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, Helvetica, Arial, sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\"",
  "typography-font-text": "system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, Helvetica, Arial, sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\"",
  "typography-root-line-height": 1.25,
  "typography-root-text-size": "14px",
}
`;
